### PR TITLE
Add CONTRIBUTING.md file.

### PR DESCRIPTION
**Pull request by rjeffman**
This PR adds a document with information on how to contribute to
ansible-freeipa development, showing the environment configuration,
available tools, and some guidelines that should be followed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+the experience on using the project much better.
+
+Also, the [playbooks] can be seen as part of the documentation, as they are
+examples of commonly performed tasks.
+
+---
+[FreeIPA]: https://freeipa.org
+[FreeIPA's Code of Conduct]: https://github.com/freeipa/freeipa/blob/master/CODE_OF_CONDUCT.md
+[for modules]: https://github.com/freeipa/ansible-freeipa/pull/357
+[for roles]: https://github.com/freeipa/ansible-freeipa/pull/423
```
Comment (t-woerner) [inline CONTRIBUTING.md:113]: Please use the combined back role PR here (430). PR 423 has not been merged...

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Upon review, it is fine to `force push` the changes.
+
+**Preparing the development environment**
+
+There are some useful tools that will help you develop for ansible-freeipa,
+and you should install, at least, the modules in `requirements.txt`:
+
+```
+python3 -m pip install --user -r requirements.txt
```
Comment (t-woerner) [inline CONTRIBUTING.md:40]: Is it requirements.txt here? The installation using pip is one possible solution. Using packages from the distribution is also possible.

File: CONTRIBUTING.md (hunk 1 above)
Comment (rjeffman) [reply CONTRIBUTING.md:113]: fixed.

File: CONTRIBUTING.md (hunk 2 above)
Comment (rjeffman) [reply CONTRIBUTING.md:40]: It should be requirements-dev.txt.

Fixed.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+Testing
+-------
+
+When testing ansible-freeipa's roles and modules, we aim to check if they
+do what they intend to do, report the results correctly, and if they are
+idempotent (although, sometimes the operation performed is not, like when
+renaming items). To achieve this, we use Ansible playbooks.
+
+The Ansible playbooks test can be found under the [tests] directory. They
```
Comment (t-woerner) [inline CONTRIBUTING.md:79]: [tests] is not defined.

**Review by t-woerner — APPROVED** [review]
LGTM

**Pull request merged**